### PR TITLE
Isogram approaches

### PR DESCRIPTION
**Pull request by bobahop**
initial commit for files and code. Explanatory details to follow...

### CODE REVIEW

**Comment by bobahop** [comment]
@ErikSchierboom Whew! Isogram approaches for Rust is now ready for review.

File: exercises/practice/isogram/.approaches/bitfield-func/snippet.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+    candidate.bytes()
+        .filter_map(|c| {
+            c.is_ascii_alphabetic()
+                .then(|| 1u32 << (c.to_ascii_lowercase() - A_LCASE))
+        })
+        .try_fold(0u32, |ltr_flags, ltr| {
+            (ltr_flags & ltr == 0).then(|| ltr_flags | ltr)
+        }).is_some()
```
Comment (ErikSchierboom) [inline exercises/practice/isogram/.approaches/bitfield-func/snippet.txt:8]:
```suggestion
candidate.bytes()
    .filter_map(|c| {
        c.is_ascii_alphabetic()
            .then(|| 1u32 << (c.to_ascii_lowercase() - A_LCASE))
    })
    .try_fold(0u32, |ltr_flags, ltr| {
        (ltr_flags & ltr == 0).then(|| ltr_flags | ltr)
    }).is_some()
```

File: exercises/practice/isogram/.approaches/config.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {
+      "uuid": "ee6b5efb-7780-45f3-ae85-9a6ba4142e57",
+      "slug": "bitfied",
+      "title": "Bit field using a for loop",
+      "blurb": "Use a bit field with a for loop to keep track of used letters.",
+      "authors": ["bobahop"]
+    },
+    {
+      "uuid": "2a1c6080-2dfc-4d5b-841e-d26b22e1d061",
+      "slug": "bitfield-func",
```
Comment (ErikSchierboom) [inline exercises/practice/isogram/.approaches/config.json:22]: You could consider being slightly more verbose with the slug, which could improve SEO for the page:

```suggestion
      "slug": "bitfield-functional",
```

or

```suggestion
      "slug": "bitfield-functionally",
```

Feel free to ignore

File: exercises/practice/isogram/.approaches/filter-all/content.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+You might want to to call the `str` method [to_ascii_lowercase][str-to-ascii-lowercase] and save calling `map`,
+like so
+
+```rust
+    candidate
+        .to_ascii_lowercase()
+        .bytes()
+        .filter(|c| c.is_ascii_alphabetic())
+        .all(|c| hs.insert(c))
```
Comment (ErikSchierboom) [inline exercises/practice/isogram/.approaches/filter-all/content.md:51]:
```suggestion
candidate
    .to_ascii_lowercase()
    .bytes()
    .filter(|c| c.is_ascii_alphabetic())
    .all(|c| hs.insert(c))
```

**Review by ErikSchierboom — APPROVED** [review]
Just a couple of tiny nits

**Comment by bobahop** [comment]
@ErikSchierboom Thanks for the suggestions! All are accepted and this is ready for another check if you'd like, or for merge.

**Pull request merged**